### DISCUSSION TITLE
Dingz Shades (Blindes) not in HomeKit

### DISCUSSION THREAD

**Discussion by matresch** [johannrichard/homebridge-dingz#531] [Q&A]
Hello, 
I am using homebridge-dingz-plugin v3.2.0, everything work fine, autdiscovery finds all my Lights, but the shades (blindes) are not in HomeKit?
- Markus

**Comment by johannrichard** [comment]
When you (re)start homebridge, the plugin should indicate in the logs what configuration it detected. Could you check if you see output similar to the following one (of course adapted to your setup, the important bit is one or more `configureWindowCoveringService() -> add Blind [Name of the blinds as set in the dingz] Blinds (M2/1)`):

```
[11/12/2022, 10:26:35 AM] [NCC-90200] [War dingz] Config changed -> will update accessory (10.0.70.79)
[11/12/2022, 10:26:35 AM] [NCC-90200] [War dingz] setDimmerConfig() -> Configuring output 'Leftie' as 'non_dimmable' (D1/0)
[11/12/2022, 10:26:35 AM] [NCC-90200] [War dingz] setDimmerConfig() -> Configuring output 'Unstoppable' as 'led' (D2/1)
[11/12/2022, 10:26:35 AM] [NCC-90200] [War dingz] configureWindowCoveringService() -> add Blind Contactless Blinds (M2/1)
[11/12/2022, 10:26:35 AM] [NCC-90200] [War dingz] Adding Button Service -> War Button 1  ->  1
[11/12/2022, 10:26:35 AM] [NCC-90200] [War dingz] Adding Button Service -> War Button 2  ->  2
[11/12/2022, 10:26:35 AM] [NCC-90200] [War dingz] Adding Button Service -> UP  ->  3
[11/12/2022, 10:26:35 AM] [NCC-90200] [War dingz] Adding Button Service -> DOWN  ->  4
```

- if you see these outputs, then the shades should be visible in HomeKit too (lights & blinds are displayed as one tile as a default, so you don't see the individual accessories from the HomeKit screen. You have to go to "Accessory details" and uncheck the "Show as Single Tile" and/or configure the blinds there)
- if you don't see the blinds being added, that likely means the configuration of the dingz is not correct, i.e. the  `DIP` switches are not set for a blind configuration. It would then be a question of configuring the dingz itself correctly, which is out of scope of the plugin (see [https://fordum.dingz.ch](https://fordum.dingz.ch) for the community support)
- if the dingz *is* configured correctly and the plugin still doesn't show the blinds/shades, we would have to dig deeper. In that case, I would ask you to provide more information, including but not limited to:
  - discovery information (i.e. the output from the logs related to the dingz, ideally while running homebride in DEBUG mode)
  - firmware versions of your dingz

**Comment by matresch** [comment]
Unfortunately not, I see no logs about detected blinds. The plug-in detects everything else (all 4 switches. lights, motion sensor, temperature sensor) but not the blinds. I have blinds installed for most Dingz and set a dip switch to Motor accordingly. Anything else I can try od maybe did wrong?

**Reply by johannrichard** [reply]
If you access the dingz via the dingz app or via the Web-UI, do you see the blinds there (choose "Remote Control"). In my case, I have two lights and one blind, configured with the DIP switch. 

![image](https://user-images.githubusercontent.com/189003/201547127-310b09aa-0531-40c3-8683-a3d14e8d554b.png)

Second, I'm afraid I have to ask again: did you try separating the tile?

I've screen-recorded how a dingz looks like in HomeKit after discovery and what you have to do to make the Bilbo's appear as individual accessories. Can you confirm you tried this?

https://user-images.githubusercontent.com/189003/201547197-c801d885-31a7-47e5-8934-36d9c6bed1f5.MOV

And last but not least: can you post the output from three dingz API endpoints here. **Please make sure you mask sensible information, namely the `MAC` fields, SSIDs or IP addresses.**

```
http://<ip address>/api/v1/device

http://<ip address>/api/v1/dimmer_config

http://<ip address>/api/v1/blind_config
```

This would allow me to check a number of things to help you debug.

**Comment by matresch** [comment]
Thanks so much for your help. Here are the answers to your questions (apologies, I haver a german DINGZ user interface, but it is the same):

 

1.	Yes I do see the blinds in the APP and Web-UI using Renote Control, see screen shot here:

 



 

2.	Yes, I confirm, I have separated the tiles in HomeKit, as you show in the video : This gives me separate tiles for lights and sensors, but no blinds

 

3.	here is the output from three dingz API endpoints here

 

api/v1/device
 

{"XXXXXXXXX":{"type":"dingz","battery":false,"reachable":true,"meshroot":true,"fw_version":"1.3.30","hw_version":"1.1.2","fw_version_puck":"1.1.29","bl_version_puck":"1.0.0","hw_version_puck":"1.1.2","hw_id_puck":65535,"puck_sn":"B20091200081","puck_production_date":{"year":20,"month":9,"day":12},"dip_config":1,"puck_hw_model":"DZ1B-4CH","front_hw_model":"dz1f-pir","front_production_date":"20/4/24","front_sn":"F20042400066","has_pir":true,"hash":"7cd63089"}}

 
 
api/v1/dimmer_config
 

{"dimmers":[{"output":"led","name":"Ausgang","feedback":null,"feedback_intensity":100},{"output":"non_dimmable","name":"Ausgang","feedback":null,"feedback_intensity":100},{"output":"halogen","name":"Ausgang","feedback":null,"feedback_intensity":100},{"output":"halogen","name":"Ausgang","feedback":null,"feedback_intensity":100}]}

 
 
api/v1/blind_config

 

{"blinds":[{"auto_calibration":true,"state":"Not initialised","invert_direction":false,"lamella_time":1.8,"shade_up_time":180,"shade_down_time":180,"type":"lamella_90","min_value":0,"max_value":100,"name":""},{"auto_calibration":true,"state":"Initialised","invert_direction":false,"lamella_time":1.8,"shade_up_time":56.46,"shade_down_time":54.83,"type":"lamella_90","min_value":0,"max_value":100,"name":""}]}

 

 

 

4.	Last, is this configuration correct ?

**Comment by johannrichard** [comment]
I guess I might have found the culprit: The "name" field of your Blinds is empty. 

The plugin currently doesn't check for an empty string (silly me!) and it could be HomeKit doesn't like empty names (but doesn't display a default name either). I could reproduce this here. The easiest potential workaround would be to set a name in the dingz ui. The plugin will pick up the change. You should then: 

- see the names displayed in the log, and 
- HomeKit will hopefully pick up these and display the blinds.

PS: if you reply by e-mail with images, those images will not end up on GitHub. I took the liberty to edit-out some spurious data from these replies.

**Reply by johannrichard** [reply]
:tada: This issue has been resolved in version 3.2.1-alpha.1 :tada: 

You can install it in HomeBridge by choosing "install alternative version".

**Comment by matresch** [comment]
Thanks fort he help. The updated version 3.2.1 now detects the blinds.

 

Unfortunately, loading the Dingz plugin now generatest he following error causing homebridge restart:

 

[…] [dingz platform] [Dingz GZ1] Config changed -> will update accessory (XXXXX)

[…] [dingz platform] [Dingz GZ1] setDimmerConfig() -> Configuring output 'Ausgang' as 'non_dimmable' (D1/0)

[…] [dingz platform] [Dingz GZ1] configureWindowCoveringService() -> add Blind 'GZ1 Storen1' (M2/1)

[…] [dingz platform] [Dingz GZ1] Adding Button Service -> Gä1-1  ->  1

[…] [dingz platform] [Dingz GZ1] Adding Button Service -> Gä1-3  ->  3

[…] [dingz platform] [Dingz GZ1] Adding Button Service -> Gä1-4  ->  4

[…] [dingz platform] Process ended, socket destroyed

[…] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null

[…] [HB Supervisor] Restarting Homebridge...

[…] [HB Supervisor] Starting Homebridge with extra flags: -I -P /var/lib/homebridge/node_modules --strict-plugin-resolution

[…] [HB Supervisor] Started Homebridge v1.5.1 with PID: XXXXX

 

 

Von: Johann Richard ***@***.***> 
Gesendet: Dienstag, 15. November 2022 21:30
An: johannrichard/homebridge-dingz ***@***.***>
Cc: matresch ***@***.***>; Author ***@***.***>
Betreff: Re: [johannrichard/homebridge-dingz] Dingz Shades (Blindes) not in HomeKit (Discussion #531)

 

I guess I might have found the culprit: The "name" field of your Blinds is empty.

The plugin currently doesn't check for an empty string (silly me!) and it could be HomeKit doesn't like empty names (but doesn't dispaly a default name either). The easiest potential workaround would be to set a name in the dingz ui and then restart homebridge. You should then:

*	see the names displayed in the log, and
*	HomeKit will hopefully pick-up these and display the blinds.

{"blinds":[{"auto_calibration":true,"state":"Not initialised","invert_direction":false,"lamella_time":1.8,"shade_up_time":180,"shade_down_time":180,"type":"lamella_90","min_value":0,"max_value":100,"name":""},{"auto_calibration":true,"state":"Initialised","invert_direction":false,"lamella_time":1.8,"shade_up_time":56.46,"shade_down_time":54.83,"type":"lamella_90","min_value":0,"max_value":100,"name":""}]}

PS: if you reply by e-mail with images, those images will not end up on GitHub. I took the liberty to edit-out some spurious data from these replies.

**Reply by johannrichard** [reply]
Glad to hear it works. These are not error messages. These are regular log messages.

**Reply by johannrichard** [reply]
In case this is not solved yet for you:

- from the extract you sent it is not clear what caused HB to stop, normally there are exceptions in the log (red log output) that hint at the cause, we would need at least these plus some context around that log output;
- depending on the cause it might helps to enable the "DEBUG" option (`-D`) to get more info what's happening and causing the crash.

On a more practical level, it would be really helpful if you could escape the logs when posting them:

````
```
log output
```
````

You can probably even do this when replying by email (not sure that works though) or, better, you post it directly here on GitHub. It
